dev/com.ibm.ws.logging.core/src/com/ibm/ws/collector/manager/buffer/BufferManagerImpl.java

```diff
@@ -22,27 +22,25 @@
 
 import com.ibm.websphere.ras.Tr;
 import com.ibm.websphere.ras.TraceComponent;
-import com.ibm.ws.ffdc.annotation.FFDCIgnore;
 import com.ibm.wsspi.collector.manager.BufferManager;
 import com.ibm.wsspi.collector.manager.SynchronousHandler;
 
 public class BufferManagerImpl extends BufferManager {
 
     private static final TraceComponent tc = Tr.register(BufferManagerImpl.class);	
     private Buffer<Object> ringBuffer;
-    private static final ReentrantReadWriteLock RERWLOCK = new ReentrantReadWriteLock(true);
+    private final ReentrantReadWriteLock RERWLOCK = new ReentrantReadWriteLock(true);
     private Set<SynchronousHandler> synchronousHandlerSet = new HashSet<SynchronousHandler>();
     private final int capacity;
 
     private final String sourceId;
     /* Map to keep track of the next event for a handler */
     private final ConcurrentHashMap<String, HandlerStats> handlerEventMap = new ConcurrentHashMap<String, HandlerStats>();
-    private static List<BufferManager> bufferManagerList= new ArrayList<BufferManager>();
+    static List<BufferManager> bufferManagerList= Collections.synchronizedList(new ArrayList<BufferManager>());
```

Comment:
We should have a quick discussion on whether the list should be in BufferManagerImpl vs EMQController. EMQController makes it cleaner to handle concurrency.